### PR TITLE
CAL-1225: User picker in Participants tab is case sensitive

### PR DESCRIPTION
**Pull request by azayati**
HibernateIdentityStoreSessionImpl and HibernateIdentityStoreImpl are added from picketlink code to gatein-portal code in order to apply the modifications.

### CODE REVIEW

**Comment by boubaker** [comment]
- You should add the original HibernateIdentityStoreSessionImpl and HibernateIdentityStoreImpl in an eXo package if possible
- Don't delete the usage of ExoHibernateIdentityStoreImpl. This class should extend the new copied one.
- Add comments to indicate the start and end of the modified / added section in forked classes
- Don't change original format of forked source code

**Comment by boubaker** [comment]
ok for me

**Comment by thomasdelhomenie** [comment]
What is OriginalHibernateIdentityStoreImpl ? The name suggests that it is just a copy of the PicketLink class but you mention in the PR that you made somme changes.
So if it is not just a copy :
* change its name (for example ExoHibernateIdentityStoreImpl)
* I would suggest to do 2 commits : one for the copy of the PicketLink class, the other ones with our changes. It would make it easy to know what we changed and apply it to newer versions
* do we really need to copy the whole class ? @azayati I thought we talked about another solution that would avoid to duplicate code.

**Comment by boubaker** [comment]
Unfortunately, ExoHibernateIdentityStoreImpl is already used. I agree that we should change the name, I suggest PatchedHibernateIdentityStoreImpl.

> I would suggest to do 2 commits : one for the copy of the PicketLink class, the other ones with our changes. It would make it easy to know what we changed and apply it to newer versions

+1.

> do we really need to copy the whole class ?

IMHO, it's better to fix this to all use cases, not only for Calendar.

**Comment by thomasdelhomenie** [comment]
Ok Boubaker, let's do it this way.

**Comment by azayati** [comment]
@thomasdelhomenie PR updated

File: web/portal/src/main/webapp/WEB-INF/conf/organization/picketlink-idm/picketlink-idm-config.xml [hunk 1]
```diff
@@ -64,7 +64,7 @@
     <identity-stores>
       <identity-store>
         <id>HibernateStore</id>
-        <class>org.picketlink.idm.impl.store.hibernate.HibernateIdentityStoreImpl</class>
+        <class>org.gatein.portal.idm.impl.store.hibernate.ExoHibernateIdentityStoreImpl</class>
```
Comment (thomasdelhomenie) [inline web/portal/src/main/webapp/WEB-INF/conf/organization/picketlink-idm/picketlink-idm-config.xml]: I thought we already used a custom version of HibernateIdentityStoreImpl, is it really useful ?

File: component/identity/src/main/java/org/picketlink/idm/impl/store/hibernate/PatchedHibernateIdentityStoreImpl.java [hunk 2]
```diff
@@ -748,15 +748,22 @@ public IdentityObject findIdentityObject(IdentityStoreInvocationContext ctx, Str
...
-         if (criteria != null && criteria.getFilter() != null)
-         {
-            queryParams.put("ioName", criteria.getFilter().replaceAll("\\*", "%"));
-         }
-         else
-         {
-            queryParams.put("ioName", "%");
-         }
+         /* BEGIN CAL-1225: User picker in Participants tab is case sensitive */
+         if (criteria != null && criteria.getFilter() != null) {
```
Comment (thomasdelhomenie) [inline component/identity/src/main/java/org/picketlink/idm/impl/store/hibernate/PatchedHibernateIdentityStoreImpl.java]: In the original class, the criteria " and io.name like :ioName" is added even when criteria of filter is null, whereas it is not the case in our implementation. Can it be an problem ?

File: web/portal/src/main/webapp/WEB-INF/conf/organization/picketlink-idm/picketlink-idm-config.xml (hunk 1 above)
Comment (boubaker) [inline web/portal/src/main/webapp/WEB-INF/conf/organization/picketlink-idm/picketlink-idm-config.xml]: For Hibernate DB store only, no. The ExoHibernateIdentityStoreImpl is used specially for DB + LDAP. So we can use both, ExoHibernateIdentityStoreImpl or PatchedHibernateIdentityStoreImpl.
> is it really useful
We have to modify this class to use PatchedHibernateIdentityStoreImpl or ExoHibernateIdentityStoreImpl. IMO, both are same, so we can use ExoHibernateIdentityStoreImpl for all cases.

**Comment by thomasdelhomenie** [comment]
Thanks.
I added comments.
About the commits, you should change the commits messages. Currently there are equals so it does not allow to know what each commit does. The first commit message should be something like "Import original PicketLink HibernateIdentityStoreImpl class in exoplatform code base for further modifications", and the second one should describe the changes done.

File: web/portal/src/main/webapp/WEB-INF/conf/organization/picketlink-idm/picketlink-idm-config.xml (hunk 1 above)
Comment (thomasdelhomenie) [inline web/portal/src/main/webapp/WEB-INF/conf/organization/picketlink-idm/picketlink-idm-config.xml]: ok I see. Thanks.

File: component/identity/src/main/java/org/picketlink/idm/impl/store/hibernate/PatchedHibernateIdentityStoreImpl.java (hunk 2 above)
Comment (azayati) [inline component/identity/src/main/java/org/picketlink/idm/impl/store/hibernate/PatchedHibernateIdentityStoreImpl.java]: Normally if the criteria is null, we should return all stored users so we don't need to add the io.name like "%" condition to the query.

**Comment by azayati** [comment]
We will modify the commit messages in the step commit -a of merge.

File: component/identity/src/main/java/org/picketlink/idm/impl/store/hibernate/PatchedHibernateIdentityStoreImpl.java (hunk 2 above)
Comment (thomasdelhomenie) [inline component/identity/src/main/java/org/picketlink/idm/impl/store/hibernate/PatchedHibernateIdentityStoreImpl.java]: ok

**Comment by thomasdelhomenie** [comment]
LGTM

**Comment by azayati** [comment]
@thomasdelhomenie it was not possible to change the commit log messages when doing the merge, I think will rervert the commits and recommit again with the good log messages.

**Comment by thomasdelhomenie** [comment]
Of course you cannot change the commit message when doing the merge, but you should have changed the commit messages before merging...
git rebase -i is fine for that for example.

Moreover I see there are 3 commits for this issue whereas the PR contains only 2 commits, what happened ?

**Comment by azayati** [comment]
We have tested the git rebase -i , the command is well executed but the commit log messages are not changed. For the 3 commits, it should be a problem with the squash.

**Comment by boubaker** [comment]
Apparently it has added my comment modification as a commit. Should we revert develop branch and attempt with another approach ? (The message was Merge...)

**Comment by thomasdelhomenie** [comment]
> We have tested the git rebase -i , the command is well executed but the commit log messages are not changed.

git rebase -i works fine for rewriting the commit messages, I do it all the time.

> Apparently it has added my comment modification as a commit. Should we revert develop branch and attempt with another approach ?

Yes we will

**Pull request merged**